uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/OneAuthTestApp.java

```diff
@@ -0,0 +1,200 @@
+
+        Logger.i(TAG, "Adding username and password on login screen..");
+        try {
+            final UiObject emailField = UiAutomatorUtils.obtainUiObjectWithTextAndClassType(
+                    "", EditText.class);
+            emailField.setText(username);
+            final UiObject nextBtn = UiAutomatorUtils.obtainUiObjectWithTextAndClassType(
+                    "Next", Button.class);
+            nextBtn.click();
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError("Could not click on object with txt Next");
+        }
+
+        Logger.i(TAG, "Handle AAD Login page prompt..");
+        // handle prompt in AAD login page
+        new AadPromptHandler(promptHandlerParameters).handlePrompt(username, password);
+    }
+
+    public void handleBackButton() {
+        UiAutomatorUtils.pressBack();
+    }
+
+    @Override
+    public void onAccountAdded() {
+        // nothing needed here
+    }
+
+    @Override
+    public void confirmAccount(@NonNull final String username) {
+        // Make sure we are seeing the output text view
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtGeneralInfo");
+        try {
+            Assert.assertTrue(resultUIObject.getText().contains("Result: Success"));
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError("Could not click on object with txt general info text");
+        }
+    }
+
+
+    public void assertSuccess() {
+        try {
+            final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtGeneralInfo");
+            resultUIObject.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+
+            final UiObject accountIdObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtAccountId");
+            accountIdObject.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+
+            Assert.assertTrue(resultUIObject.getText().contains("Result: Success"));
+            Assert.assertFalse(TextUtils.isEmpty(resultUIObject.getText()));
+            Assert.assertFalse(TextUtils.isEmpty(accountIdObject.getText()));
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError("Could not click on object general text and account id");
+        }
+    }
+
+    /**
+     * Returns string list of all accounts available
+     */
+    public List<String> getAllAccounts() {
+        List<String> accountsList = new ArrayList<>();
+        UiAutomatorUtils.handleButtonClick("com.microsoft.oneauth.testapp:id/get_all_accounts_button");
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/all_accounts_list");
+        try {
+            int size = resultUIObject.getChildCount();
+            for (int i = 0; i < size; i++) {
```

Comment:
```suggestion
            for (int i = 0; i < resultUIObject.getChildCount(); i++) {
```

Reply:
Not a major difference in both the things, also resultUIObject.getChildCount() internally has couple of lines that need to be executed, it would be better to execute them once instead of executing them everytime inside the for loop check. Agreed, instead of size, I can rename the variable to childCount.